File: sol/1072/1072.cpp
#include <unordered_map>

struct Solution {
    using vb = vector<bool>;
    int maxEqualRowsAfterFlips(vector<vector<int>>& matrix) {
        int m = matrix[0].size(), res = 0;
        vb buff(m);
        unordered_map<vb, int> d;
        for (auto &r : matrix){
            if (r[0] == 0){
                res = max(res, ++d[vb(r.begin(), r.end())]);
            } else {
                for (int i = 0; i < m; ++i){
                    buff[i] = r[i] ^ 1;
                }
                res = max(res, ++d[buff]);
            }
        }
        return res;
    }
};

File: sol/1072/1072.py
from collections import Counter

class Solution:
    def maxEqualRowsAfterFlips(self, matrix: List[List[int]]) -> int:
        d = Counter()
        for r in matrix:
            if r[0] == 0:
                d[tuple(r)] += 1
            else:
                d[tuple(1 ^ x for x in r)] += 1
        return max(d.values())
